src/WebJobs.Script/OutOfProc/Http/HttpWorkerDescription.cs

```diff
@@ -1,15 +1,31 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
+using System.IO;
 
 namespace Microsoft.Azure.WebJobs.Script.OutOfProc.Http
 {
     public class HttpWorkerDescription : WorkerDescription
     {
-        public override void ApplyDefaultsAndValidate()
+        public override void ApplyDefaultsAndValidate(string workerDirectory)
         {
-            base.ApplyDefaultsAndValidate();
+            if (workerDirectory == null)
+            {
+                throw new ArgumentNullException(nameof(workerDirectory));
+            }
+            Arguments = Arguments ?? new List<string>();
+            WorkerDirectory = WorkerDirectory ?? workerDirectory;
+            if (string.IsNullOrEmpty(DefaultWorkerPath) && !string.IsNullOrEmpty(DefaultExecutablePath) && !Path.IsPathRooted(DefaultExecutablePath))
```

Comment:
Can you explain the logic here? Why are we evaluating `DefaultWorkerPath` here?

What are the scenarios where `WorkerDirectory` won't be `null` and overriden with the value provided in the argument?

Comments here would help a bit.

Reply:
- If DefaultWorkerPath is not set then compute full path for DefaultExecutablePath from scriptRootDir

- If DefaultWorkerPath is set and find full path from scriptRootDir

Will add comments